include/autoppl/expression/binop.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <type_traits>
+#include <autoppl/util/traits.hpp>
+#include <autoppl/expression/var_expr.hpp>
+#include <autoppl/expression/variable.hpp>
+
+namespace ppl {
+
+template <class LHSValueType, class RHSValueType, class BinaryOp>
+struct BinaryOpNode 
+{
+	using binop_result_t = std::common_type_t<
+		typename node_traits<LHSValueType>::binop_result_t,
```

Comment:
I think you can just use `value_t` instead of adding a new alias `binop_result_t` for the interface. Seems like they alias the same thing in `Variable`. In my current branch, `node_traits` has been replaced with `var_expr_traits`.